File: wondes_journey_code_csharp/Attack/Attack.cs
public class Attack
{
    private int element;
    public int damage;
    private string elementName;
    private int baseDamage;
    private int elementDamage;
    private int criticDamage;
    private int criticProb;
    private int missProb;
    private bool criticAttack;
    private bool elementalAttack;

    public Attack(
        int element,
        int baseDamage,
        int elementDamage,
        int cricticDamage,
        int criticProb,
        int missProb)
    {
        this.element = element;
        this.baseDamage = baseDamage;
        this.elementDamage = elementDamage;
        this.criticDamage = cricticDamage;
        this.criticProb = criticProb;
        this.missProb = missProb;
        this.criticAttack = false;
        this.elementalAttack = false;
    }

    public void SetElementName()
    {
        if (this.element == 1)
        {
            this.elementName = "FIRE";
            this.elementalAttack = true;
        }

        else if (this.element == 2)
        {
            this.elementName = "WATER";
            this.elementalAttack = true;
        }

        else if (this.element == 3)
        {
            this.elementName = "ICE";
            this.elementalAttack = true;
        }

        else
        {
            this.elementName = "NORMAL";
            this.elementalAttack = false;
        }
    }

    public bool ItsCritic(int numberForProbs)
    {
        if (numberForProbs <= this.criticProb)
        {
            this.criticAttack = true;
        }

        else
        {
            this.criticAttack = false;
        }

        return this.criticAttack;
    }

    public bool AttackMissed(int numberForProbs)
    {
        if (numberForProbs <= this.missProb)
        {
            return true;
        }

        else
        {
            return false;
        }
    }

    public void CalcAttackDamage()
    {
        Random rand = new Random();
        if (this.AttackMissed(rand.Next(0, 100)) == false)
        {
            this.damage = this.baseDamage;

            if (this.element == 1 || this.element == 2 || this.element == 3)
            {
                this.damage += this.elementDamage;
            }

            if (this.ItsCritic(rand.Next(0, 100)) == true)
            {
                this.damage += this.criticDamage;
                Console.WriteLine(" ");
                Console.WriteLine("¡EL ATAQUE HA SIDO CRÍTICO!");
                Console.WriteLine(" ");
            }
        }

        else
        {
            this.damage = 0;
            Console.WriteLine(" ");
            Console.WriteLine("¡EL ATAQUE HA FALLADO!");
            Console.WriteLine(" ");
        }

        Console.WriteLine("¡EL ATAQUE INFLIGIDO ", this.damage, " DE DAÑO!");
        Console.WriteLine(" ");
    }
}

File: wodens_journey_game/Attack/Attack.py
import random

class Attack:

    def __init__(self, element, base_damage, element_damage, critic_damage, critic_prob, miss_prob):
        
        self.element = element
        self.base_damage = base_damage
        self.element_damage = element_damage
        self.critic_damage = critic_damage
        self.critic_prob = critic_prob
        self.miss_prob = miss_prob
        self.element_name = ""
        self.damage = 0
        self.critic_attack = False
        self.elemental_attack = False
    
    def get_element_name(self):

        if self.element == 1:
            
            self.element_name = "FIRE"
            self.elemental_attack = True
        
        elif self.element == 2:
            
            self.element_name = "WATER"
            self.elemental_attack = True
        
        elif self.element == 3:
            
            self.element_name = "ICE"
            self.elemental_attack = True
        
        else:
            
            self.element_name = "NORMAL"
            self.elemental_attack = False
    
    def its_critic(self, number_for_probs):

        if number_for_probs <= self.critic_prob:
            
            self.critic_attack = True
        
        else:
            
            self.critic_attack = True
        
        return self.critic_attack
    
    def attack_missed(self, number_for_probs):

        if number_for_probs <= self.miss_prob:

            return True
        
        else:
            
            return False
    
    def calc_damage_attack(self):

        if not self.attack_missed(random.randint(0, 100)):

            self.damage = self.base_damage

            if self.element == 1 or self.element == 2 or self.element == 3:
                
                self.damage += self.element_damage
            
            if self.its_critic(random.randint(0, 100)):
                
                self.damage += self.critic_damage
                print(" ")
                print("¡EL ATAQUE HA SIDO CRÍTICO!")
                print(" ")
        
        else:

            self.damage = 0
            print(" ")
            print("¡EL ATAQUE HA FALLADO!")
            print(" ")
        
        print(f"¡EL ATAQUE HA INFLIGIDO {self.damage} DE DAÑO!")